mirrord-layer/src/lib.rs

```diff
@@ -88,12 +88,43 @@ pub(crate) static ENABLED_UDP_OUTGOING: OnceLock<bool> = OnceLock::new();
 
 pub(crate) const MIRRORD_SKIP_LOAD: &str = "MIRRORD_SKIP_LOAD";
 
+/// Check if we're running in NixOS or Devbox
+/// if so, add `sh` to the skip list because of https://github.com/metalbear-co/mirrord/issues/531
+fn nix_devbox_patch(config: &mut LayerConfig) {
+    let mut current_skip = config
+        .skip_processes
+        .clone()
+        .map(VecOrSingle::to_vec)
+        .unwrap_or_default();
+    // already contains, doesn't matter
+    if current_skip.contains(&"sh".to_string()) {
+        return;
+    }
+
+    if is_nix_or_devbox() {
```

Comment:
nit-2: to drop the `return`.

```suggestion
    if is_nix_or_devbox() && !current_skip.contains(...) {
```